Add clarifications around untyped collections (#1953)
oasis-tcs/odata-specs: odata-json-format/7 Structural Property.md

```
:::

> A primitive value within an untyped collection of primitive values

Reviewer: ```suggestion
A primitive value within an untyped collection
```
Can untyped collections contain a mix of primitive, object, and array values?

Reviewer: Why have this here?

Better move it to section "Untyped Value" and link to this section here.

Author: Moved.

The value of a property of type `Collection(Edm.Untyped)`MUST
be a collection, and it MAY contain any combination of primitive values,

> structural values, and collections. Enum values within an untyped 
> collection SHOULD be represented as a string, using the member name.

Reviewer 2: ```suggestion
structural values, and collections. Enumeration values within an untyped 
collection SHOULD be represented as a string, using the `enumerationMember`.
```
Use the same wording as in section 7.1.

Reviewer: In other specs we only use the long form "enumeration value".

Author: fixed.

control information, in which case they MUST conform to the type
described by the control information.

> Collections within an untyped collection are themselves untyped 

Reviewer 2: ```suggestion
Collections directly contained within an untyped collection are themselves untyped 
```

Reviewer: Isn't the collection value of an object property in an untyped collection also untyped?

If so, the "directly" would mislead me.

Reviewer 2: Can't I specify `@type` control information even in an untyped collection?

Reviewer: 🤷🏻‍♂️

Author: a structured item within the untyped collection could have a collection-valued property whose @type was specified. so "directly contained" is more precise.

Author: Renamed to "directly contained"
